android-emulator-plugin/src/main/java/com/quittle/androidemulator/AndroidEmulatorPlugin.java

```diff
@@ -153,9 +153,15 @@ private static void createInstallEmulatorSystemImageTask(final Project project,
     private static void createCreateEmulatorTask(final Project project, final EmulatorConfiguration emulatorConfiguration) {
         final File avdRoot = emulatorConfiguration.getAvdRoot();
         final String emulatorName = emulatorConfiguration.getEmulatorName();
+        final String emulatorDevice = emulatorConfiguration.getDeviceType();
         createExecTask(project, emulatorConfiguration, CREATE_ANDROID_EMULATOR_TASK_NAME, exec -> {
                     exec.setExecutable(emulatorConfiguration.getAvdManager());
-                    exec.setArgs(l("create", "avd", "--name", emulatorName, "--package", emulatorConfiguration.getSystemImagePackageName(), "--force"));
+                    List<String> args = l("create", "avd", "--name", emulatorName, "--package", emulatorConfiguration.getSystemImagePackageName(), "--force");
+                    if (emulatorDevice != null) {
+                        args.add("--device");
```

Comment:
I was able to reproduce the issue in CI and see the stacktrace by running `./gradlew -p example-android-project --stacktrace`

```
Caused by: java.lang.UnsupportedOperationException
        at com.quittle.androidemulator.AndroidEmulatorPlugin.lambda$createCreateEmulatorTask$8(AndroidEmulatorPlugin.java:161)
        at com.quittle.androidemulator.AndroidEmulatorPlugin.lambda$createExecTask$22(AndroidEmulatorPlugin.java:305)
        at org.gradle.api.internal.tasks.DefaultTaskContainer.create(DefaultTaskContainer.java:366)
        at com.quittle.androidemulator.AndroidEmulatorPlugin.createExecTask(AndroidEmulatorPlugin.java:302)
        at com.quittle.androidemulator.AndroidEmulatorPlugin.createCreateEmulatorTask(AndroidEmulatorPlugin.java:157)
        at com.quittle.androidemulator.AndroidEmulatorPlugin.lambda$apply$0(AndroidEmulatorPlugin.java:63)
        ...
```

I believe the issue is that my helper function `l()` uses `Arrays.asList()` which returns an immutable List so you can't just add an entry to it, hence the `UnsupportedOperationException`. I recommend updating the helper function to build and return a mutable list instead.

Reply:
Nice catch. I made it an ArrayList.